### PR TITLE
feat: Add customer managed key to encrypt resources

### PR DESCRIPTION
**Pull request by kayman-mk**
Instead of using the default AWS keys, using `encryption_key_id` allows a customer managed key to encrypt all resources created by the module.

### CODE REVIEW

**Comment by baolsen** [comment]
Related: #18

File: main.tf [hunk 1]
```diff
@@ -71,5 +71,5 @@ module "billing_notifier_lambda" {
   }
 
   vpc_config = local.vpc_config
-
+  kms_key_id = local.lambda_encryption_key_id
```
Comment (baolsen) [inline main.tf]: FYI, `kms_key_id` is used for encrypting cloudwatch logs within the `lambda-auto-package` module. It is unfortunately not a clear name. 

It means the value of `/alias/aws/lambda` wont work here.

The module now also supports `lambda_kms_key_arn` for encrypting the lambda environment variables (related #18). 

As cloudwatch logs are encrypted by default, I suggest we change this to something like:

```
kms_key_id = var.encryption_key_id
lambda_kms_key_arn = local.lambda_encryption_key_id
```

File: variables.tf [hunk 2]
```diff
@@ -48,3 +48,9 @@ variable "subnet_ids" {
   type        = list(string)
   default     = []
 }
+
```
Comment (baolsen) [inline variables.tf]: Suggest name this variable: `kms_key_id` to fit with other naming

**Comment by kayman-mk** [comment]
I close this one. Seems that was fixed in the meantime.

**Pull request closed without merging**